mysql-test/t/set_thread_var.test

```diff
@@ -0,0 +1,148 @@
+SET SESSION max_join_size = 2002;
+
+--let $assert_text = Check if max_join_size is correct
+--let $assert_cond = @@max_join_size = 2002
+--source include/assert.inc
+
+
+--connection user1
+
+--let $assert_text = Check if max_join_size is correct
+--let $assert_cond = @@max_join_size = 1001
+--source include/assert.inc
+
+
+# Reconnect as root
+--connection default
+
+# Set variable for both current thd and user1 from root thd
+--replace_regex /SESSION [0-9]* max_join_size = 1002, SESSION [0-9]* max_join_size = 2003/SESSION $conn_mysqluser1 max_join_size = 1002, SESSION $conn_root max_join_size = 2003/
+--eval SET SESSION $conn_mysqluser1 max_join_size = 1002, SESSION $conn_root max_join_size = 2003
+
+--let $assert_text = Check if max_join_size is correct
+--let $assert_cond = @@max_join_size = 2003
+--source include/assert.inc
+
+
+--connection user1
+
+--let $assert_text = Check if max_join_size is correct
+--let $assert_cond = @@max_join_size = 1002
+--source include/assert.inc
+
+# Set variable for root thd from user1 thd
+--replace_regex /SESSION [0-9]*/SESSION $conn_root/ /thread [0-9]*/thread $root/
+--error ER_SHOW_DENIED_ERROR
+--eval SET SESSION $conn_root max_join_size = 2004
+
+# Set variable for user1 thd from user1 thd
+--replace_regex /SESSION [0-9]*/SESSION $mysqluser1/
+--eval SET SESSION $conn_mysqluser1 max_join_size = 1003
+
+--let $assert_text = Check if max_join_size is correct
+--let $assert_cond = @@max_join_size = 1003
+--source include/assert.inc
+
+# Set variable for user1 thd from user1 thd without session ID
+SET SESSION max_join_size = 1004;
+
+--let $assert_text = Check if max_join_size is correct
+--let $assert_cond = @@max_join_size = 1004
+--source include/assert.inc
+
+
+--connection default
+
+--let $assert_text = Check if max_join_size is correct
+--let $assert_cond = @@max_join_size = 2003
+--source include/assert.inc
+
+# Set multiple variables for both current thd and user1 from root thd
+--replace_regex /SESSION [0-9]* max_join_size = 1005/SESSION $conn_mysqluser1 max_join_size = 1005/ /SESSION [0-9]* max_join_size = 2004/SESSION $conn_root max_join_size = 2004/
+--eval SET SESSION $conn_mysqluser1 max_join_size = 1005, max_error_count = 30, SESSION $conn_root max_join_size = 2004, max_error_count = 31
+
+--let $assert_text = Check if max_join_size is correct
+--let $assert_cond = @@max_join_size = 2004 AND @@max_error_count = 31
+--source include/assert.inc
+
+
+--connection user1
+
+--let $assert_text = Check if max_join_size is correct
+--let $assert_cond = @@max_join_size = 1005 AND @@max_error_count = 30
+--source include/assert.inc
+
+--error ER_NO_SUCH_THREAD
+--eval SET SESSION 999 max_join_size = 9999
+
+--error ER_PARSE_ERROR
+--eval SET GLOBAL 1 max_join_size = 1000
+
+--error ER_PARSE_ERROR
+--eval SET SESSION -1 max_join_size = 1000
+
+--error ER_PARSE_ERROR
+--eval SET SESSION 1 TRANSACTION READ WRITE
+
+--connection default
+SET GLOBAL TRANSACTION READ ONLY;
```

Comment:
`SET SESSION [THREAD_ID] TRANSACTION` returns `ER_PARSE_ERROR` with both this and original patches. We can add `SET SESSION [THREAD_ID] TRANSACTION` but this will be an additional feature.